core/node/groups.go

```diff
@@ -337,16 +337,26 @@ func Online(bcfg *BuildCfg, cfg *config.Config, userResourceOverrides rcmgr.Part
 
 	isBitswapLibp2pEnabled := cfg.Bitswap.Libp2pEnabled.WithDefault(config.DefaultBitswapLibp2pEnabled)
 	isBitswapServerEnabled := cfg.Bitswap.ServerEnabled.WithDefault(config.DefaultBitswapServerEnabled)
+	isHTTPRetrievalEnabled := cfg.HTTPRetrieval.Enabled.WithDefault(config.DefaultHTTPRetrievalEnabled)
 
-	// Don't provide from bitswap when the legacy noop experiment "strategic provider service" is active
-	isBitswapServerEnabled = isBitswapServerEnabled && !cfg.Experimental.StrategicProviding
+	// Right now Provider and Reprovider systems are tied together - disabling Reprovider by setting interval to 0 disables Provider
+	// and vice versa: Provider.Enabled=false will disable both Provider of new CIDs and the Reprovider of old ones.
+	isProviderEnabled := cfg.Provider.Enabled.WithDefault(config.DefaultProviderEnabled) && cfg.Reprovider.Interval.WithDefault(config.DefaultReproviderInterval) != 0
```

Comment:
When `Reprovide.Interval` is set to `0` do we want to 1) only disable periodic reprovides, while leaving initial provide and manual reprovides enabled or 2) disable DHT providing altogether?

Reply:
I believe we want to disable provider here because many users set `Reprovide.Interval`  set to `0` to stop providing. 

If we don't disable it, then many users will suddenly become providers for 48h.

Imo manual provides are not something many people use without also having regular provide and reprovide system enabled – what I've seen is people running everything + often abused DHT by doing extra manual provides in a tight loop.

Note: times changed. We now have useful `Reprovider.Strategy` which allow users to fine-tune what is reprovided without having to disable entire system. We should discourage disabling entire system, and encourage fine-tuning what is reprovided.